cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -556,27 +618,32 @@ static void sc_mount_nvidia_driver_multiarch(const char *rootfs_dir, const char
 
 static void sc_mount_vulkan(const char *rootfs_dir)
 {
-	const char *vulkan_sources[] = {
-		SC_VULKAN_SOURCE_DIR,
-	};
-	const size_t vulkan_sources_len =
-	    sizeof vulkan_sources / sizeof *vulkan_sources;
+	sc_mkdir_and_mount_and_glob_files(rootfs_dir, SC_VULKAN_SOURCE_DIR, NULL, SC_VULKAN_DIR, vulkan_globs, vulkan_globs_len);
+}
 
-	sc_mkdir_and_mount_and_glob_files(rootfs_dir, vulkan_sources,
-					  vulkan_sources_len, SC_VULKAN_DIR,
-					  vulkan_globs, vulkan_globs_len);
+static void sc_mount_glvnd(const char *rootfs_dir)
+{
+	sc_mkdir_and_mount_and_glob_files(rootfs_dir, SC_GLVND_VENDOR_SOURCE_DIR, NULL, SC_GLVND_DIR, glvnd_vendor_globs, glvnd_vendor_globs_len);
 }
 
 static void sc_mount_egl(const char *rootfs_dir)
 {
-	const char *egl_vendor_sources[] = { SC_EGL_VENDOR_SOURCE_DIR };
-	const size_t egl_vendor_sources_len =
-	    sizeof egl_vendor_sources / sizeof *egl_vendor_sources;
-
-	sc_mkdir_and_mount_and_glob_files(rootfs_dir, egl_vendor_sources,
-					  egl_vendor_sources_len, SC_GLVND_DIR,
-					  egl_vendor_globs,
-					  egl_vendor_globs_len);
+	sc_mkdir_and_mount_and_glob_files(rootfs_dir, SC_EGL_VENDOR_SOURCE_DIR, NULL, SC_GLVND_DIR, egl_vendor_globs, egl_vendor_globs_len);
+	setenv("__EGL_VENDOR_LIBRARY_DIRS", SC_GLVND_DIR "/egl_vendor.d", true);
+	setenv("__EGL_EXTERNAL_PLATFORM_CONFIG_DIRS", SC_GLVND_DIR "/egl_external_platform.d", true);
+}
+
+// This is all a dirty hack involving mangling the snap's filesystem.
+// Unfortunately, desktop-launcher from snapcraft unconditionally sets
+// LIBVA_DRIVERS_PATH, which means that we have no choice except to put any
```

Comment:
Would it be better to fix snapcraft to not set this unconditionally rather than having snapd try and workaround it?